climada/hazard/tc_tracks_synth.py

```diff
@@ -166,6 +241,219 @@ def _one_rnd_walk(track, ens_size, ens_amp0, ens_amp, max_angle, rnd_vec):
+    x_next = 2 * np.sqrt(3) * (_f_ac(np.cos(theta) * x + np.sin(theta) * y, gamma) - 1 / 2)
+    return x_next
+
+
+@jit
+def _f_ac(z, theta):
+    """
+    F transform for autocorrelated random uniform series generation
+
+    Implements function F defined in:
+    https://stats.stackexchange.com/questions/48086/
+        algorithm-to-produce-autocorrelated-uniformly-distributed-number
+    i.e., the CDF of Y.
+
+    Parameters
+    ----------
+    z : float
+        Value.
+    theta : float
+        arccos of autocorrelation.
+
+    Returns
+    -------
+        res : float
+            CDF at value z
+    """
+    c = np.cos(theta)
+    s = np.sin(theta)
+    if z >= np.sqrt(3) * (c + s):
+        res = 1
+    elif z > np.sqrt(3) * (c - s):
+        res = 1 / 12 / np.sin(2 * theta) * \
+              (-3 - z ** 2 + 2 * np.sqrt(3) * z * (c + s) + 9 * np.sin(2 * theta))
+    elif z > np.sqrt(3) * (-c + s):
+        res = 1 / 6 * (3 + np.sqrt(3) * z / c)
+    elif z > -np.sqrt(3) * (c + s):
+        res = 1 / 12 / np.sin(2 * theta) * \
+              (z ** 2 + 2 * np.sqrt(3) * z * (c + s) + 3 * (1 + np.sin(2 * theta)))
+    else:
+        res = 0
+    return res
+
+
+@jit
+def _get_bearing_angle(lon, lat):
+    """
+    Compute bearing angle of great circle paths defined by consecutive points
+
+    Returns initial bearing (also called forward azimuth) of the n-1 great circle
+    paths define by n consecutive longitude/latitude points. The bearing is the angle
+    (clockwise from North) which if followed in a straight line along a great-circle
+    arc will take you from the start point to the end point. See also:
+    http://www.movable-type.co.uk/scripts/latlong.html
+    Here, the bearing of each pair of consecutive points is computed.
+
+    Parameters
+    ----------
+    lon : numpy.ndarray of shape (n,)
+        Longitude coordinates of consecutive point, in decimal degrees.
+    lat : numpy.ndarray of shape (n,)
+        Latitude coordinates of consecutive point, in decimal degrees.
+
+    Returns
+    -------
+        earth_ang_fix : numpy.ndarray of shape (n-1,)
+            Bearing angle for each segment, in decimal degrees
+    """
+    lon, lat = map(np.radians, [lon, lat])
+    lat_1 = lat[:-1]
+    lon_1 = lon[:-1]
+    lat_2 = lat[1:]
+    lon_2 = lon[1:]
+    delta_lon = lon_2 - lon_1
+    # what to do with the points that don't move?
+    #   i.e. where lat_2=lat_1 and lon_2=lon_1? The angle does not matter in
+    # that case because angular distance will be 0.
+    earth_ang_fix = np.arctan2(np.sin(delta_lon) * np.cos(lat_2),
+                               np.cos(lat_1) * np.sin(lat_2) - \
+                               np.sin(lat_1) * np.cos(lat_2) * np.cos(delta_lon))
+    return np.degrees(earth_ang_fix)
+
+
+@jit
+def _get_angular_distance(lon, lat):
+    """
+    Compute the angular distance of great circle paths defined by consecutive points
```

Comment:
It's true that `climada.util.coordinates.dist_approx` returns the distance in km, while here it's in degrees. But that's only a factor. I would modify `dist_approx` by adding an option to compute the angular or metric distance.

Reply:
I can do and started to, but `climada.util.coordinates.dist_approx ` compute distances between all pairs of points while here we only need those between consecutive points. So using that function would imply more computations than required, hence I would prefer not to change that. What do you think?

Reply:
No extra computations are required if you replace the call
```
angular_dist = _get_angular_distance(lon, lat)
```
by
```
angular_dist = dist_approx(lat[:-1, None], lon[:-1, None], lat[1:, None], lon[1:, None],
                           method="geosphere", units="degree")[:, 0, 0]
```
The `units` parameter needs to be implemented and should default to `"km"` for backwards compatibility.

Reply:
Ok, did so in commits 0f02c64574d3acf39b8a68d6136336f3d711ab2e and 5e5623dbf2fd45f28ff529911aca21589c6cfb29

Reply:
Of course one could also think that functions `_get_bearing_angle` and `_get_destination_points` could be in `climada.util.coordinates`, but I would say they are too specific.

Reply:
Thanks! I added unit tests for the new `units` parameter in 569526105fe96fa8452604a3f3dcfb485f732f4d.

Reply:
All references to the now-removed function `_get_angular_distance` should be removed. Especially in the unit tests.

Reply:
Oh yes. Did in 256cc3711b8ac598633814a6c40904a4fe67d3e2